kindtest/runner_test.go

```diff
@@ -381,6 +389,14 @@ func testRunner() {
 			return nil
 		}).ShouldNot(HaveOccurred())
 
+		Eventually(func() error {
+			_, _, err := kubectl("get", "secret", "-n", runner2NS, "runner-token-"+runnerPool2Name)
+			if err == nil {
```

Comment:
You should check whether err is NotFound or not.
If there is an error in api-server, then this test will pass.

Reply:
Fixed the test to check whether NotFound or not from the output result of kubectl command.